src/smd-core.c

```diff
@@ -1018,7 +1030,10 @@ static int smd_attr_copy_val_to_internal(char *out, smd_dtype_t *t, smd_dtype_t
         case (SMD_TYPE_UINT8): {
           uint8_t ov = *(uint8_t*)val;
           // check the range
-          if(ov < FLT_MIN || ov > FLT_MAX){
+          if(ov > 0 && (ov < FLT_MIN || ov > FLT_MAX)){
+            return 1;
+          }
+          if(ov < 0 && (ov > -FLT_MIN || ov < -FLT_MAX)){
```

Comment:
Unsigned types cannot compare less than zero. They simply do not contain a sign bit. As such, any `ov < 0` check with an unsigned `ov` type is as pointless as checking `a*a < 0`: The check just won't be successful for any real number.

Again, systematic problem, I won't repeat myself.

Reply:
More redundant tests, ok. It's just easier to implement all the tests and then remove the ones that don't make sense. I didn't remove them yet because not all the conversions are tested.

Reply:
Well, my main issue is with the *form* of the tests, which does not make any sense at all: When you try to check whether an integer value is in range for a floating point type, checking against `FLT_MIN` is absolutely pointless. That constant should never appear in code that does a range check. You are doing a total of six comparisons, four logical operations, and two separate `if()` statements where a simple `if(abs(ov) > FLT_MAX)` would have sufficed.